src/utilities/core/Deprecated.hpp

```diff
@@ -57,4 +57,26 @@
 #  define OS_DEPRECATED
 #endif
 
+// Adapted from: https://www.fluentcpp.com/2019/08/30/how-to-disable-a-warning-in-cpp/
+// clang-format off
+#if defined(_MSC_VER)
+
+#  define DISABLE_WARNING_PUSH __pragma(warning(push))
+#  define DISABLE_WARNING_POP __pragma(warning(pop))
+#  define DISABLE_WARNING(warningNumber) __pragma(warning(disable : warningNumber))
+
+#  define DISABLE_WARNING_DEPRECATED DISABLE_WARNING(4996)
+
+#elif defined(__GNUC__) || defined(__clang__)
+
+#  define DO_PRAGMA(X) _Pragma(#X)
+#  define DISABLE_WARNING_PUSH DO_PRAGMA(GCC diagnostic push)
+#  define DISABLE_WARNING_POP DO_PRAGMA(GCC diagnostic pop)
+#  define DISABLE_WARNING(warningName) DO_PRAGMA(GCC diagnostic ignored #warningName)
+
+#  define DISABLE_WARNING_DEPRECATED DISABLE_WARNING(-Wdeprecated-declarations)
+
+#endif
+// clang-format on
```

Comment:
New macro for clarity/shortness. See it in example at https://godbolt.org/z/jEKzcedoK across 3 compilers

Reply:
I removed it, swig was choking on it